marlowe-hspec/test/Spec.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Main (main) where
+
+import           Language.Marlowe.Semantics (AccountId (..), Action (..), Case (..), Contract (..), Payee (..),
+                                             Token (..), Value (..))
+import           Test.Hspec                 (Selector, describe, hspec, it, shouldThrow)
+import           Test.Hspec.Marlowe         (assertNoWarnings)
+import           Test.HUnit.Lang            (FailureReason (..), HUnitFailure (..))
+
+warningException :: Selector HUnitFailure
+warningException (HUnitFailure _ (ExpectedButGot Nothing "Right Nothing" "Right (Just (Slot {getSlot = 5},[TransactionInput {txInterval = (Slot {getSlot = 6},Slot {getSlot = 11}), txInputs = [IDeposit (AccountId 1 (PubKey {getPubKey = 706172747931})) (PubKey {getPubKey = 706172747931}) (Token  ) 100]},TransactionInput {txInterval = (Slot {getSlot = 8},Slot {getSlot = 14}), txInputs = [IDeposit (AccountId 2 (PubKey {getPubKey = 706172747932})) (PubKey {getPubKey = 706172747932}) (Token  ) 300]}],[TransactionPartialPay (AccountId 1 (PubKey {getPubKey = 706172747931})) (Party (PubKey {getPubKey = 706172747932})) (Token  ) 100 500]))")) = True
```

Comment:
This will break any time anyone changes anything in the ledger library. No checked in literals that aren't golden tests which can be updated with `--accept`!

Also, maybe you want to use `Pretty` for some of these things.

Reply:
I wanted something more specific than `(/=) "Right Nothing"` but since it's a string that's a little annoying. Perhaps I should `read` the string and then I could pick out important things like `isRight` and `isJust` together.

Reply:
If you want to have a big string, put it in a file and use `tasty-golden`. Then you can run the test executable with `--accept` and it updates the expected files.

Reply:
`read` was a disaster so I've given things a little more user friendly failure message and just checked that the bit that won't ever change is still there. It's very likely that the other part will randomly stop working because apart from anything else it depends on the Z3 implementation running for the exact counter examples it finds.